oepcie/yuanphil_impl/oe_ctx.h

```diff
@@ -0,0 +1,31 @@
+#ifndef _INCL_GUARD_ONE
+#define _INCL_GUARD_ONE
+
+#include "oe_ctx_impl.h"
+
+typedef struct oe_ctx_impl *oe_ctx;
+
+int oe_init(oe_ctx* ctx);
+
+oe_ctx* oe_create();
+
+void oe_destroy(oe_ctx* ctx);
+
+int oe_set_opt(oe_ctx* state, int option, const char* value);
+
+/*
+ * For functional purposes, here we are going to assume that value points to
+ * NULL, and we will allocate the string to return. It is up to the use to cl-
+ * ean up memory.
+ */
+int oe_get_opt(oe_ctx* state, int option, char* value);
+
+int oe_write_reg(oe_ctx* state, int device_id, int addr, int value);
```

Comment:
spec calls for `state` to be `const`. This may not be possible because elements of `state` are mutated by this operation. Let me know if I should change the spec. Second arg should be `size_t device_idx` , which an  index within the device map, not an id which specifies device type. This should resolve the issue about having two or more of the same device mentioned in the comments.